source/configuration_manual/authentication/oauth2.rst

```diff
@@ -258,60 +271,112 @@ Oauth2 overrides some of the default HTTP client and SSL settings:
+   You only need this if the username that comes from authentication will not otherwise
+   match with :dovecot_core:ref:`oauth2_username_attribute` value.
 
-  ## How introspection is made, valid values are
-  ##   auth = GET request with Bearer authentication
-  ##   get  = GET request with token appended to URL
-  ##   post = POST request with token=bearer_token as content
-  ##   local = Attempt to locally validate and decode JWT token
-  # introspection_mode = auth
+.. dovecot_core:setting:: oauth2_active_attribute
+   :values: @string
 
-  ## Force introspection even if tokeninfo contains wanted fields
-  ## Set this to yes if you are using active_attribute
-  # force_introspection = no
+   Attribute name for checking whether account is disabled. (optional)
 
-  ## Validation key dictionary, turns on local validation
-  # local_validation_key_dict =
+.. dovecot_core:setting:: oauth2_active_value
+   :values: @string
 
-  ## A space separated list of scopes of validity (optional)
-  # scope = something
+   Expected value in active_attribute. (empty = require present, but anything goes)
 
-  ## username attribute in response (default: email)
-  # username_attribute = email
+.. dovecot_core:setting:: oauth2_issuers
+   :values: @boollist
 
-  ## username normalization format (default: %Lu)
-  # username_format = %Lu
+   Valid issuer(s) for the token.
 
-  ## Attribute name for checking whether account is disabled (optional)
-  # active_attribute =
+.. dovecot_core:setting:: oauth2_openid_configuration_url
+   :values: @string
 
-  ## Expected value in active_attribute (empty = require present, but anything goes)
-  # active_value =
+   URL to :rfc:`7628` OpenID Provider Configuration Information schema.
 
-  ## Expected issuer(s) for the token (space separated list)
-  # issuers =
+.. dovecot_core:setting:: oauth2_fields
+   :values: @strlist
 
+   Key-value fields to include in successful authentication.
 
-  ## URL to RFC 7628 OpenID Provider Configuration Information schema
-  # openid_configuration_url =
+.. dovecot_core:setting:: oauth2_send_auth_headers
+   :values: @boolean
+   :default: no
 
-  ## Extra fields to set in passdb response (in passdb static style)
-  # pass_attrs =
+   Whether to send special headers about authentication to remote server.
+   If you enable this, following headers will be sent:
 
-  ## Timeout in milliseconds
-  # timeout_msecs = 0
+   ``X-Dovecot-Auth-Protocol``
+      Requested protocol, such as imap or pop3.
+   ``X-Dovecot-Auth-Local``
+      Local IP address where client connected to.
+   ``X-Dovecot-Auth-Remote``
+      Remote IP address of the client connection.
 
-  ## Enable debug logging
-  # debug = no
+.. dovecot_core:setting:: oauth2_use_worker
+   :default: yes
+   :values: @boolean
 
-  ## Use worker to verity token
-  # blocking = yes
+   Use worker process to verity token, or perform password grant request.
```

Comment:
* Typo verity.
 * Clarify that it does all the introspection etc etc lookups in worker process.
 * Point to passdb_use_worker and say it must be the same if passdb oauth2 is used.